src/runtime.js

```diff
@@ -0,0 +1,38 @@
+;(function(){
+
+  function Runtime(context){
+    this.context = context;
+  };
+
+  function BrowserRuntime(context){
+    this.context = context;
+  }
+
+  function NonBrowserRuntime(context){
+    this.context = context;
+  }
+
+  Runtime.prototype.getWindow = function() {
+    return this.context;
+  };
+
+  BrowserRuntime.prototype = new Runtime();
+  BrowserRuntime.prototype.constructor = BrowserRuntime;
+
+  BrowserRuntime.prototype.getDocument = function(){
+    return this.context.document;
+  }
+
+  NonBrowserRuntime.prototype = new Runtime();
+  NonBrowserRuntime.prototype.constructor = NonBrowserRuntime;
+
+  NonBrowserRuntime.prototype.getDocument = function(){
+    return this.context;
+  }
+
+  this._PusherRuntimes = {
```

Comment:
Ah, these don't live on the `Pusher` global. But they probably should as we don't want to pollute the global namespace. Also see my comment about `runtime.js` coming before/after `pusher.js`. Probably a few things that need to be considered here.